plugins/destination/s3/client/write.go

```diff
@@ -68,11 +68,12 @@ func (c *Client) WriteTable(ctx context.Context, msgs <-chan *message.WriteInser
 			table := msg.GetTable()
 			objKey := c.spec.ReplacePathVariables(table.Name, uuid.NewString(), time.Now().UTC(), c.syncID)
 			// if object was already initialized, use the same key
-			// We don't need any locking here because all messages for the same table are processed sequentially
```

Comment:
Why is this assumption not true anymore? Or was it never true?

Reply:
Looking at the batch code (even before https://github.com/cloudquery/plugin-sdk/pull/1719) I'm not sure we ever had a guarantee that `WriteTable` is not called from multiple Go routines